Gems/RecastNavigation/Code/Source/Misc/RecastHelpers.h

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+#include <DetourAlloc.h>
+#include <Recast.h>
+#include <AzCore/Math/Aabb.h>
+#include <AzCore/Math/Vector3.h>
+
+namespace RecastNavigation
+{
+    //! A helper class to manage different coordinate systems between O3DE and Recast Navigation library.
+    //! In O3DE, +Z is up. In Recast library, +Y is up.
+    class RecastVector3
+    {
+    public:
+        RecastVector3() = default;
+
+        //! A constructor from O3DE coordinate values.
+        explicit RecastVector3(const AZ::Vector3& in)
+        {
+            m_x = in.GetX();
+            m_y = in.GetZ(); // swapping y and z
+            m_z = in.GetY();
+        }
+
+        //! A constructor from Recast coordinate values.
+        explicit RecastVector3(const float* data)
+        {
+            m_x = data[0];
+            m_y = data[1];
+            m_z = data[2];
```

Comment:
question: To match the behavior of the `AZ::Vector3` constructor should this be switched to be:

```
m_x = data[0];
m_y = data[2];
m_z = data[1];
```

? 🤔 This might also be confusing so perhaps we should have an explicit static function that is called `CreateFromFloatSwapYZ` or something? I know it's a bit long-winded, but better to be clear and unambiguous

Reply:
I have a command on this constructor:

`//! A constructor from Recast coordinate values.`

There are two ways here: one from O3DE->Recast and from Recast to O3DE. And sometimes it needs to stays in Recast format while working with Recast API. So I default the internal state to be Recast friendly.

Reply:
But I like the idea of helper methods.

Reply:
Reworked to use helper static methods here to clarity.